controllers/courses.js

```diff
@@ -620,99 +622,122 @@ router.get('/:courseId/', (req, res, next) => {
+		const userHasEditorEnabled = EDITOR_URL && edtrUser;
+		const courseHasNewEditorLessons = ((_newLessons || {}).total || 0) > 0;
+
+		const isNewEdtrioActivated = editorBackendIsAlive && (courseHasNewEditorLessons || userHasEditorEnabled);
+		// ################################ end new Editor check ##################################
+
+		const ltiToolIds = (course.ltiToolIds || []).filter(
+			ltiTool => ltiTool.isTemplate !== 'true',
+		).map((tool) => {
+			tool.isBBB = tool.name === 'Video-Konferenz mit BigBlueButton';
+			return tool;
+		});
+		const lessons = (_lessons.data || []).map(lesson => Object.assign(lesson, {
+			url: `/courses/${req.params.courseId}/topics/${lesson._id}/`,
+		}));
 
-			homeworks.sort((a, b) => {
-				if (a.dueDate > b.dueDate) {
-					return 1;
-				}
-				return -1;
-			});
+		const homeworks = (_homeworks.data || []).map((assignment) => {
+			assignment.url = `/homework/${assignment._id}`;
+			return assignment;
+		});
 
-			const baseUrl = (req.headers.origin || process.env.HOST || 'http://localhost:3100');
-			const courseGroups = permissionHelper.userHasPermission(
-				res.locals.currentUser,
-				'COURSE_EDIT',
-			)
-				? _courseGroups.data || []
-				: (_courseGroups.data || []).filter(cg => cg.userIds.some(
-					user => user._id === res.locals.currentUser._id,
-				));
-
-			// ###################### start of code for new Editor ################################
-			let newLessons;
-			if (isNewEdtrioActivated) {
-				newLessons = (_newLessons.data || []).map(lesson => ({
-					...lesson,
-					url: `/courses/${req.params.courseId}/topics/${lesson._id}?edtr=true`,
-					hidden: !lesson.visible,
-				}));
+		homeworks.sort((a, b) => {
+			if (a.dueDate > b.dueDate) {
+				return 1;
 			}
+			return -1;
+		});
 
-			// ###################### end of code for new Editor ################################
-			res.render(
-				'courses/course',
-				Object.assign({}, course, {
-					title: course.isArchived
-						? `${course.name} (archiviert)`
-						: course.name,
-					activeTab: req.query.activeTab,
-					lessons,
-					homeworks: homeworks.filter(task => !task.private),
-					myhomeworks: homeworks.filter(task => task.private),
-					ltiToolIds,
-					courseGroups,
-					baseUrl,
-					breadcrumb: [
-						{
-							title: 'Meine Kurse',
-							url: '/courses',
-						},
-						{
-							title: course.name,
-							url: `/courses/${course._id}`,
-						},
-					],
-					filesUrl: `/files/courses/${req.params.courseId}`,
-					nextEvent: recurringEventsHelper.getNextEventForCourseTimes(
-						course.times,
-					),
-					// #################### new Editor, till replacing old one ######################
-					newLessons,
-					isNewEdtrioActivated,
-					scopedCoursePermission: scopedPermissions[res.locals.currentUser._id],
-				}),
-			);
-		})
-		.catch(next);
+		const baseUrl = req.headers.origin || HOST || 'http://localhost:3100';
```

Comment:
move this (`HOST || 'http://localhost:3100'`) to config/globals.js or use HOST only

Reply:
I do not add it and i do not know for what origin in context of CSRF is used. It is adding by aron.